src/components/SingleProductPage/SingleProductPage.jsx

```diff
@@ -0,0 +1,201 @@
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { toast } from "react-toastify";
+import { useParams } from "react-router";
+import { Breadcrumb } from "../../components";
+import { ICONS_URL } from "../../constants";
+import {
+  useLoader,
+  useCart,
+  useAuth,
+  useAuthModal,
+  useWishlist,
+} from "../../context";
+import {
+  addToCartHandler,
+  addToWishlistHandler,
+  isInArrayList,
+  removeFromWishlistHandler,
+  getSingleProductHandler,
+} from "../../utilities";
+import styles from "./SingleProductPage.module.css";
+
+const SingleProductPage = () => {
+  const { productID } = useParams();
+  const { setLoader } = useLoader();
+  const { authModalHandler } = useAuthModal();
+  const { authState } = useAuth();
+  const { cartState, cartDispatch } = useCart();
+  const { wishlistState, wishlistDispatch } = useWishlist();
+  const [product, setProduct] = useState([]);
+
+  const [itemAdded, setItemAdded] = useState(
+    isInArrayList(product, cartState.cart)
+  );
+  const [isWishlisted, setIsWishlisted] = useState(
+    isInArrayList(product, wishlistState.wishlist)
+  );
+
+  useEffect(() => {
+    setLoader(true);
+    (async () => {
+      try {
+        const productFromResponse = await getSingleProductHandler(productID);
```

Comment:
check the response status before updating the state , you havent checked for status in handler as well as in service

Reply:
Sure, will resolve it in the next PR.